activerecord/lib/active_record/persistence.rb

```diff
@@ -215,7 +215,7 @@ def becomes(klass)
       became.instance_variable_set("@changed_attributes", attributes_changed_by_setter)
       became.instance_variable_set("@new_record", new_record?)
       became.instance_variable_set("@destroyed", destroyed?)
-      became.instance_variable_set("@errors", errors)
+      became.errors.copy!(errors)
```

Comment:
This literally just needs to be `became.instance_variable_set("@errors", errors.dup)`